driver/receiver.cpp

```diff
@@ -26,6 +26,11 @@ Receiver::Receiver(const std::string& device_uid,
     roc_context_config net_context_config;
     memset(&net_context_config, 0, sizeof(net_context_config));
 
+    if (device_encoding.channel_layout == ROC_CHANNEL_LAYOUT_MULTITRACK && device_encoding.channel_count > 4) {
+        net_context_config.max_packet_size = device_encoding.channel_count * 512;
+        net_context_config.max_frame_size = device_encoding.channel_count * 1024;
+    }
```

Comment:
Two suggestions:

1. Extract this into `max_packet_size()` and `max_frame_size()` methods of `DeviceLocalEncoding` and reuse them in sender and receiver.

2. Rework formulas roughly like this:

	```
    max_frame_size = channel_count * 1024 * 4
    max_packet_size = channel_count * 1024 * 4 + 128
	```

	Here, 1024 is max samples per packet, 4 is max bytes per sample, and 128 is max packet header size.

(Let me now if this approach works on your setup)